packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -41,11 +47,24 @@ const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
 				'yyyy MM dd'
 			)}
 			className={classNames}
-			disabled={day.outside}
-			onClick={() => onClick(day.date)}
+			disabled={outside}
+			onClick={() => onClick(date)}
+			onKeyDown={(event) => {
+				// When tabbing and selecting a DayNumber using
+				// SPACE key the active state it's not being removed.
+				// See https://github.com/liferay/clay/issues/3374 for more details.
+				if (event.key === KEYS.SPACE) {
```

Comment:
@diegonvs we should probably also check out "Spacebar" here.

> [1] Older browsers may return "Spacebar" instead of " " for the Space Bar key. Firefox did so until version 37, as did Internet Explorer 9, 10, and 11.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

Reply:
![spacebar](https://user-images.githubusercontent.com/905006/86385181-29c04380-bc90-11ea-90c1-75ab27c683b7.jpg)

https://www.threadless.com/product/2574/spacebar

Reply:
See 64af3da